src/sagemaker/fw_utils.py

```diff
@@ -1066,31 +1074,43 @@ def validate_torch_distributed_distribution(
         return
 
     err_msg = ""
+    # Check instance compatibility
+    if not _is_gpu_instance(instance_type):
+        err_msg += "torch_distributed is supported only for GPU instances.\n"
```

Comment:
Same as below (Trainium is not a GPU instance), this needs to be updated to something like https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/fw_utils.py#L761-L763.

Reply:
updated